internal/webhook/v1beta2/integrationtestscenario_webhook.go

```diff
@@ -0,0 +1,98 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	appstudiov1beta2 "github.com/konflux-ci/integration-service/api/v1beta2"
+)
+
+// nolint:unused
+// log is for logging in this package.
+var integrationtestscenariolog = logf.Log.WithName("integrationtestscenario-resource")
+
+// SetupIntegrationTestScenarioWebhookWithManager registers the webhook for IntegrationTestScenario in the manager.
+func SetupIntegrationTestScenarioWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).For(&appstudiov1beta2.IntegrationTestScenario{}).
+		WithValidator(&IntegrationTestScenarioCustomValidator{}).
+		Complete()
+}
+
+// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+// NOTE: The 'path' attribute must follow a specific pattern and should not be modified directly here.
+// Modifying the path for an invalid path can cause API server errors; failing to locate the webhook.
+// +kubebuilder:webhook:path=/validate-appstudio-redhat-com-v1beta2-integrationtestscenario,mutating=false,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=integrationtestscenarios,verbs=create;update,versions=v1beta2,name=vintegrationtestscenario-v1beta2.kb.io,admissionReviewVersions=v1
+
+// IntegrationTestScenarioCustomValidator struct is responsible for validating the IntegrationTestScenario resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type IntegrationTestScenarioCustomValidator struct {
+	// TODO(user): Add more fields as needed for validation
+}
+
+var _ webhook.CustomValidator = &IntegrationTestScenarioCustomValidator{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type IntegrationTestScenario.
+func (v *IntegrationTestScenarioCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	integrationtestscenario, ok := obj.(*appstudiov1beta2.IntegrationTestScenario)
+	if !ok {
+		return nil, fmt.Errorf("expected a IntegrationTestScenario object but got %T", obj)
+	}
+	integrationtestscenariolog.Info("Validation for IntegrationTestScenario upon creation", "name", integrationtestscenario.GetName())
+
+	// TODO(user): fill in your validation logic upon object creation.
```

Comment:
Should this file contain the validations that are now located in https://github.com/konflux-ci/integration-service/blob/main/api/v1beta2/integrationtestscenario_webhook.go?
@jsztuka @14rcole and @Josh-Everett adding you since you both worked on the webhook validations.

Reply:
I don't have a good answer here as my exposure to webhook validations was [relatively minor](https://github.com/konflux-ci/integration-service/commit/4fbaffc0e13e7c699bfd037c601bf767ec042fa9) I would defer to @14rcole or @jsztuka

Reply:
After verifying in local testing that the admission validation will work the same way as before, I'm fine to close this discussion.